subworkflows/local/uncompress_asset/main.nf

```diff
@@ -3,38 +3,74 @@
 //
 
 include { GUNZIP as GUNZIP_FASTA } from '../../../modules/nf-core/gunzip'
-include { GUNZIP as GUNZIP_GTF   } from '../../../modules/nf-core/gunzip'
 include { GUNZIP as GUNZIP_GFF   } from '../../../modules/nf-core/gunzip'
+include { GUNZIP as GUNZIP_GTF   } from '../../../modules/nf-core/gunzip'
+include { UNTAR as UNTAR_CHR_DIR } from '../../../modules/nf-core/untar'
+include { UNZIP as UNZIP_ALLELES } from '../../../modules/nf-core/unzip'
+include { UNZIP as UNZIP_GC      } from '../../../modules/nf-core/unzip'
+include { UNZIP as UNZIP_LOCI    } from '../../../modules/nf-core/unzip'
+include { UNZIP as UNZIP_RT      } from '../../../modules/nf-core/unzip'
 
 workflow UNCOMPRESS_ASSET {
     take:
-    fasta // channel: [meta, fasta]
-    gff   // channel: [meta, gff]
-    gtf   // channel: [meta, gtf]
+    ascat_alleles // channel: [meta, ascat_alleles]
+    ascat_loci    // channel: [meta, ascat_loci]
+    ascat_loci_gc // channel: [meta, ascat_loci_gc]
+    ascat_loci_rt // channel: [meta, ascat_loci_rt]
+    chr_dir       // channel: [meta, chr_dir]
+    fasta         // channel: [meta, fasta]
+    gff           // channel: [meta, gff]
+    gtf           // channel: [meta, gtf]
 
     main:
     versions = Channel.empty()
 
-    // Do not run GUNZIP_* if the condition is false
+    // Do not run GUNZIP_*, UNTAR_*, UNZIP_* if the condition is false
+    ascat_alleles = ascat_alleles.map { meta, ascat_alleles_ -> meta.decompress_ascat_alleles ? [meta, ascat_alleles_] : null }
+    ascat_loci = ascat_loci.map { meta, ascat_loci_ -> meta.decompress_ascat_loci ? [meta, ascat_loci_] : null }
+    ascat_loci_gc = ascat_loci_gc.map { meta, ascat_loci_gc_ -> meta.decompress_ascat_loci_gc ? [meta, ascat_loci_gc_] : null }
+    ascat_loci_rt = ascat_loci_rt.map { meta, ascat_loci_rt_ -> meta.decompress_ascat_loci_rt ? [meta, ascat_loci_rt_] : null }
+    chr_dir = chr_dir.map { meta, chr_dir_ -> meta.decompress_chr_dir ? [meta, chr_dir_] : null }
     fasta = fasta.map { meta, fasta_ -> meta.decompress_fasta ? [meta, fasta_] : null }
     gff = gff.map { meta, gff_ -> meta.decompress_gff ? [meta, gff_] : null }
     gtf = gtf.map { meta, gtf_ -> meta.decompress_gtf ? [meta, gtf_] : null }
```

Comment:
Harshil Alignment

Reply:
language server is not happy with Hathil align